src/modules/claims/claims.service.ts

```diff
@@ -261,28 +262,61 @@ export class ClaimsService {
     return this._cacheClient.issueClaim(this._signerService.did, message);
   }
 
+  /**
+   *
+   * @param token optional token containing claimType, version and subject
+   * @returns claim params obtained from token
+   */
+  private getClaimTypeFromToken(token?: string) {
+    if (token) {
+      const { claimData, sub } = this._didRegistry.jwt.decode(token) as {
+        claimData: { claimType: string; claimTypeVersion: string };
+        sub: string;
+      };
+      return { ...claimData, subject: sub };
+    }
+    return {};
+  }
+
   /**
    * @description Registers issued onchain claim with Claim manager
    *
-   * @param claimId - id of signed onchain claim
+   * @param claim - id of signed onchain claim.
+   * @param token - @deprecated use subject, claimType, claimTypeVersion instead. Token should get removed and subject, claimType, claimTypeVersion should be required.
    */
-  async registerOnchain(
-    claim: Pick<Claim, 'token' | 'subjectAgreement' | 'onChainProof' | 'acceptedBy'>
-  ) {
-    if (!readyToBeRegisteredOnchain(claim)) {
-      throw new Error(ERROR_MESSAGES.CLAIM_WAS_NOT_ISSUED);
+  async registerOnchain(claim: {
+    claimType?: string;
+    claimTypeVersion?: string;
+    token?: string;
+    subjectAgreement?: string;
+    onChainProof?: string;
```

Comment:
`onChainProof` is required for onchain registration

Reply:
yes , onChainProof and acceptedBy changed to required.